setup.py

```diff
@@ -165,5 +167,5 @@ def _decode(x):
          neural networks and machine learning techniques.""",
     cmdclass=dict(build_ext=CMakeBuild),
     zip_safe=False,
-    install_requires=["numpy>=1.16", "cmake>=3.10.3", "scipy>=1.2.1", "mpi4py>=3.0.1"],
+    install_requires=["numpy>=1.16", "cmake>=3.12", "scipy>=1.2.1", "mpi4py>=3.0.1"],
```

Comment:
too new cmake?

Reply:
We had this discussion at some point.
I enabled in `setup.py` the `--parallel` flag when you call `cmake build`. But the flag is only available in cmake>=3.12 (which is ~ 9 months more recent than 3.10.3).

`--parallel` is somewhat equivalent to the `-- -j 4` that you recently merged, but has the additional nice thing that it picks automatically the number of processors and if some build system does not like -j 4 he adapts to it.

I can revert this, though, if you prefer.

Reply:
So you want this to stay, or you want this to go?

Reply:
I think we can leave it to 3.12 !